### ISSUE TITLE
Iterating months is broken

### ISSUE THREAD

**Issue by icambron** [icambron/twix.js#56]
```js
startDate = moment("2014-10-01")
endDate = moment("2014-12-01")
dateRange = moment.twix(startDate, endDate, true)

dateRange.count("months") //=> 3
it = dateRange.iterate("months")
it.next() //=> moment for 2014-10-01
it.next() //=> moment for 2014-11-01
it.next() //=> null
```

Should give us 12/1

**Comment by icambron** [comment]
Fixed in 3674ea7ef0c3 and released in 0.6.2. /cc @DouweM

**Comment by DouweM** [comment]
Awesome!
